sp-sim/src/sidecar.rs

```diff
@@ -201,6 +224,31 @@ impl Inner {
     }
 }
 
+async fn handle_request<'a>(
+    handler: &mut Handler,
+    recv: Result<(&[u8], SocketAddr)>,
+    server: &'a mut SpServer,
+    responsiveness: Responsiveness,
+    port_num: usize,
```

Comment:
as above (do we want to share this code or is it different enough that that would be more trouble than it's worth?)

Reply:
It took a tiny bit of work to make the two `handle_request()`s not-different, but most of what it needed was worth doing for its own sake (threading the sending address and receiving port through `handler.dispatch()`). Thanks for the nudge